ugs-platform/ugs-platform-plugin-setup-wizard/src/main/java/com/willwinder/ugs/nbp/setupwizard/panels/WizardPanelImportSettings.java

```diff
@@ -0,0 +1,184 @@
+        fileInfoPanel.setForeground(ThemeColors.LIGHT_GREY);
+
+        labelNameValue = new JLabel("");
+        labelNameValue.setFont(labelNameValue.getFont().deriveFont(Font.BOLD, 16));
+
+        labelFirmware = new JLabel("Firmware:", JLabel.RIGHT);
+        labelFirmware.setFont(labelFirmware.getFont().deriveFont(Font.BOLD));
+        labelFirmwareValue = new JLabel("");
+
+        labelCreatedBy = new JLabel("Created by:", JLabel.RIGHT);
+        labelCreatedBy.setFont(labelCreatedBy.getFont().deriveFont(Font.BOLD));
+        labelCreatedByValue = new JLabel("");
+
+        labelDate = new JLabel("Date:", JLabel.RIGHT);
+        labelDate.setFont(labelDate.getFont().deriveFont(Font.BOLD));
+        labelDateValue = new JLabel("");
+
+        labelSettingsImported = new JLabel("<html><body>" +
+                "<h3>Finished imported settings</h3>" +
+                "<p>Please continue to verify your configuration using the following steps.</p>" +
+                "</body></html>");
+        labelSettingsImported.setVerticalAlignment(SwingConstants.CENTER);
+        labelSettingsImported.setIcon(ImageUtilities.loadImageIcon("icons/checked32.png", false));
+
+        buttonOpen = new JButton("Open settings file...");
+        buttonOpen.addActionListener(event -> {
+            JFileChooser fileChooser = FirmwareSettingsFileTypeFilter.getSettingsFileChooser();
+            int returnVal = fileChooser.showOpenDialog(new JFrame());
+            if (returnVal == JFileChooser.APPROVE_OPTION) {
+                loadSettingsFile(fileChooser.getSelectedFile());
+            }
+        });
+
+        buttonImport = new JButton("Import settings");
+        buttonImport.addActionListener(event -> importSettings());
+    }
+
+    private void importSettings() {
+        String message = "Are you sure you want to overwrite the current firmware settings?";
+        int result = JOptionPane.showConfirmDialog(this.getComponent(), message,
+                "Overwrite firmware settings?", JOptionPane.YES_NO_OPTION);
+        if (result == JOptionPane.OK_OPTION) {
+            try {
+                getBackend().getController().getFirmwareSettings().setSettings(firmwareSettingsFile.getSettings());
+            } catch (FirmwareSettingsException e) {
+                LOGGER.log(Level.SEVERE, "Couldn't set firmware settings", e);
+            }
+            labelSettingsImported.setVisible(true);
+        }
+
+        loadSettingsFile(null);
+    }
+
+    private void loadSettingsFile(File file) {
+        if (file == null) {
+            firmwareSettingsFile = null;
+        } else {
+            try {
+                firmwareSettingsFile = FirmwareSettingUtils.readSettings(file);
+            } catch (IOException e) {
+                JOptionPane.showMessageDialog(this.getComponent(), "Couldn't read settings file: " + e.getMessage(), "Error while reading settings file", JOptionPane.ERROR_MESSAGE);
```

Comment:
There's a helper in `GUIHelper.displayErrorDialog` that I've been using to wrap dialogs. It's main purpose is to wrap it with `invokeLater`, so not strictly necessary here since it will already be tied to a swing event thread.

Reply:
It's nice to have it consistent, thanks!